.github/workflows/dispatch.yml

```diff
@@ -51,18 +58,87 @@ jobs:
-        # As such, we'll end up with one copy-workflow parallel job per target.
+        # The triggered copy workflow runs use that final array as their matrix.
+        # Since max-parallel here is 1, we'll end up with at most max-parallel from copy workflow + 1 parallel jobs.
+        # 20 is the upper limit on parallel jobs on a free plan.
         cfg: ${{ fromJson(needs.matrix.outputs.batches) }}
-    name: Start copy workflow (batch ${{ matrix.cfg.key }})
+      max-parallel: 1
+    env:
+      GITHUB_TOKEN: ${{ secrets.WEB3BOT_GITHUB_TOKEN }}
+      WORKFLOW_YML: copy-workflow.yml
+      WORKFLOW_REPO: protocol/.github
     steps:
-      - uses: benc-uk/workflow-dispatch@4c044c1613fabbe5250deadc65452d54c4ad4fc7 # v1.1.0
-        with:
-          workflow: "Deploy" # "name" attribute of copy-workflow.yml
-          token: ${{ secrets.WEB3BOT_GITHUB_TOKEN }}
-          # double toJson on matrix.cfg.value is here on purpose
-          inputs: '{ "head_commit_url": ${{ toJson(github.event.head_commit.url) }}, "targets": ${{ toJson(toJson(matrix.cfg.value)) }} }'
+      - id: dispatch
+        name: Dispatch copy workflow
+        run: |
+          start_date="$(date +%s)"
+          # 2 toJson calls are needed to turn the array of targets into a string
+          echo '{"targets":${{ toJson(toJson(matrix.cfg.value)) }}}' | jq -c '.' | gh workflow run "${{ env.WORKFLOW_YML }}" --ref "${{ github.ref }}" --repo "${{ env.WORKFLOW_REPO }}" --json
+          echo "::set-output name=start_date::$start_date"
+      - id: run
+        name: Wait for copy workflow run to start
+        run: |
+          # checks every 3 seconds until the most recent copy workflow run's created_at is later than this job's start_date
+          while sleep 3; do
+            run="$(gh api "/repos/${{ env.WORKFLOW_REPO }}/actions/workflows/${{ env.WORKFLOW_YML }}/runs?per_page=1" --jq '.workflow_runs[0]')"
+            # nothing to check if no copy workflow run was returned
+            if [[ ! -z "$run" ]]; then
+              run_start_date="$(date --date="$(jq -r '.created_at' <<< "$run")" +%s)"
+              if [[ "$run_start_date" > "${{ steps.dispatch.outputs.start_date }}" ]]; then
+                echo "::set-output name=id::$(jq -r '.id' <<< "$run")"
+                break
+              fi
+            fi
+          done
+      - name: Wait for copy workflow run to complete
+        run: |
+          # delays checking copy workflow's run status to save on GH API requests
+          sleep ${{ env.WORKFLOW_COMPLETION_CHECK_DELAY }}
+
+          # checks every 3 seconds until the copy workflow run's status is completed
+          # redirects the stdout to /dev/null because it is very chatty
+          gh run watch "${{ steps.run.outputs.id }}" --repo "${{ env.WORKFLOW_REPO }}" > /dev/null
+      - id: jobs
+        name: Find copy workflow jobs that pushed changes
+        env:
+          STEP_NAME: 'Force push web3-bot/sync branch'
```

Comment:
nitpick:
in the copy-workflow, the name is `'Force push ${{ env.GITHUB_USER }}/sync branch'`,
so if you change the `GITHUB_USER` in one place, you break the workflow in another place.
Could we comment & warn about the coupling in copy-workflow?
Could we enforce a unique name? Like "Force push unified-ci branch"?

Reply:
Good point, I made this change - https://github.com/protocol/.github/pull/261/commits/d9cfe7cab23dc3650e4925f93d4755103fbc9caa - so that it's not suggested that the username is configurable.